apps/snitch_core/lib/core/data/schema/order_address.ex

```diff
@@ -0,0 +1,123 @@
+defmodule Snitch.Data.Schema.OrderAddress do
+  @moduledoc false
+
+  use Snitch.Data.Schema
+
+  alias Snitch.Data.Schema.{Country, State}
+  alias Snitch.Repo
+
+  @required_fields ~w(first_name last_name address_line_1 city zip_code country_id)a
+  @cast_fields ~w(phone alternate_phone state_id)a ++ @required_fields
+
+  embedded_schema do
+    field(:first_name, :string)
+    field(:last_name, :string)
+    field(:address_line_1, :string)
+    field(:address_line_2, :string)
+    field(:city, :string)
+    field(:zip_code, :string)
+    field(:phone, :string)
+    field(:alternate_phone, :string)
+
+    field(:state_id, :integer)
+    field(:country_id, :integer)
+  end
+
+  def changeset(%__MODULE__{} = address, params) do
+    address
+    |> cast(params, @cast_fields)
+    |> validate_required(@required_fields)
+    |> validate_length(:address_line_1, min: 10)
+    |> assoc_state_and_country()
+  end
+
+  defp struct_to_map(struct) do
```

Comment:
We can get rid of this now, no call sites.

Reply:
yeah! :+1:

Reply:
Naming is hard, and since the body is just 2 lines, move them to the pipe at L48. :metal:
You can even consider `Map.drop/2`.